File: PY/find_num.py
def second_smallest(arr):

    smallest = float('inf')
    second = float('inf')

    for num in arr:

        if num < smallest:
            second = smallest
            smallest = num

        elif smallest < num < second:
            second = num

    if second == float('inf'):
        return -1

    return second


arr = [8,5,2,9,5,6]

print(second_smallest(arr))

File: JAVA/find_num.java
public class find_num {

    public static int secondSmallest(int[] arr){

        int smallest = Integer.MAX_VALUE;
        int second = Integer.MAX_VALUE;

        for(int num : arr){

            if(num < smallest){

                second = smallest;
                smallest = num;

            }else if(num > smallest && num < second){

                second = num;
            }
        }

        if(second == Integer.MAX_VALUE)
            return -1;

        return second;
    }

    public static void main(String[] args){

        int[] arr = {8,5,2,9,5,6};

        System.out.println(secondSmallest(arr));
    }
}